pkg/controllers/managementuser/rbac/namespace_handler.go

```diff
@@ -87,6 +90,10 @@ func (n *nsLifecycle) Updated(obj *v1.Namespace) (runtime.Object, error) {
 }
 
 func (n *nsLifecycle) Remove(obj *v1.Namespace) (runtime.Object, error) {
+	if obj.Status.Phase == v1.NamespaceTerminating {
+		n.asyncCleanupRBAC(obj.Name)
```

Comment:
Q: Does this survives rancher restart? If not for some reason, this significantly ups the chances of having orphaned cluster roles.

Reply:
Just tried this out.  It did survive the restart.  When Rancher came backup, the namespace termination continued and the rbac was cleaned-up.